CurlWrapper.php

```diff
@@ -518,10 +550,34 @@ public function setDefaultOptions()
             CURLOPT_AUTOREFERER    => true,
             CURLOPT_CONNECTTIMEOUT => 15,
             CURLOPT_TIMEOUT        => 30,
+            CURLOPT_HEADERFUNCTION => array($this, 'headerCallback'),
         );
     }
 
     /**
+     * Parses raw HTTP headers to array
+     */
+    private function parseHeaders($raw_headers) {
+        $headers = [];
+        foreach (explode("\n", $raw_headers) as $i => $h) {
+            $h = explode(':', $h, 2);
+            if (isset($h[1])) {
+                $headers[$h[0]] = trim($h[1]);
```

Comment:
What about several headers with the same name? Couple of `Set-Cookie` for example.

There is `http_parse_headers` function in `pecl_http` extension , probably we should try use it and if it's not available then emulate it, I think it would be a better approach.